lib/game_team.rb

```diff
@@ -37,6 +41,58 @@ def self.winningest_coach
   def self.worst_coach
   end
 
+  def self.game_team_shots_goals_count(arr_games)
+    season = arr_games.first.game_id
+    self.find_by(season)
+  end
+
+  def self.get_goal_shots_by_game_team(game_teams)
+    #passes in an array of games
+    results = {}
+    #each game, iterating through and adding the team_id as the key and
+    #getting the shots and goals count
+    game_teams.each do |game_team|
+      results[game_team.team_id] ||= {"shots"=>0,"goals"=>0}
+      results[game_team.team_id]["shots"] += game_team.shots
+      results[game_team.team_id]["goals"] += game_team.goals
+    end
+    results
+  end
+
+  def self.least_accurate_team(season)
+   seasonal_hash = gets_team_shots_goals_count(season)
+   seasonal_hash.map do |key,value|
+     value["average"] = (value["shots"]/ value["goals"].to_f).round(2)
+   end
+   team_hash_with_highest_average = seasonal_hash.min_by do |key,value|
+     value["average"]
+   end
+   team_hash_with_highest_average[0]
+ end
+
+ def self.most_accurate_team(season)
+    seasonal_hash = gets_team_shots_goals_count(season)
+    seasonal_hash.map do |key,value|
+      value["average"] = (value["shots"]/ value["goals"].to_f).round(2)
+    end
+    team_hash_with_highest_average = seasonal_hash.max_by do |key,value|
+      value["average"]
+    end
+    team_hash_with_highest_average[0]
+  end
+
+  def self.gets_team_shots_goals_count(season)
+    #passes in desired season, grabs the *games* for the season
+    season_games = Game.grouped_by_season(season)
+    matches = []
+    #gets the game_teams associated with the season
+    season_games.each do |game|
+      matches.concat(GameTeam.find_by(game.game_id))
+    end
```

Comment:
I think this is a great use of data gathering between 2 csv files.

Reply:
Agreed!